cmd/pr-reminder/main.go

```diff
@@ -0,0 +1,344 @@
+
+	users, err := createUsers(c, gtk, slackClient)
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to create users")
+	}
+
+	ghClient, err := o.GitHubOptions.GitHubClient(false)
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to create github client")
+	}
+
+	for _, user := range findPrsForUsers(users, c.Repos, ghClient) {
+		if len(user.PrRequests) > 0 {
+			// sort by most recent update first
+			sort.Slice(user.PrRequests, func(i, j int) bool {
+				return user.PrRequests[i].LastUpdated.After(user.PrRequests[j].LastUpdated)
+			})
+
+			if err = messageUser(user, slackClient); err != nil {
+				logrus.WithError(err).Fatal("failed to message users")
+			}
+		}
+	}
+}
+
+func findPrsForUsers(users map[string]user, repos []string, ghClient prClient) map[string]user {
+	for _, orgRepo := range repos {
+		split := strings.Split(orgRepo, "/")
+		org, repo := split[0], split[1]
+
+		prs, err := ghClient.GetPullRequests(org, repo)
+		if err != nil {
+			logrus.Errorf("failed to get pull requests: %v", err)
+		}
+		for _, pr := range prs {
+			for i, u := range users {
+				if u.requestedToReview(pr) {
+					u.PrRequests = append(u.PrRequests, prRequest{
+						Repo:        orgRepo,
+						Number:      pr.Number,
+						Url:         pr.HTMLURL,
+						Title:       pr.Title,
+						Author:      pr.User.Login,
+						Created:     pr.CreatedAt,
+						LastUpdated: pr.UpdatedAt,
+					})
+					users[i] = u
+				}
+			}
+		}
+	}
+
+	return users
+}
+
+func loadConfig(filename string, config interface{}) error {
+	configData, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return fmt.Errorf("failed to read config: %v", err)
+	}
+	if err = yaml.Unmarshal(configData, &config); err != nil {
+		return fmt.Errorf("failed to unmarshall config: %v", err)
+	}
+	return nil
+}
+
+func createUsers(config config, gtk githubToKerberos, slackClient slackClient) (map[string]user, error) {
+	users := make(map[string]user, len(config.TeamMembers))
+	for _, member := range config.TeamMembers {
+		email := fmt.Sprintf("%s@redhat.com", member)
+		slackUser, err := slackClient.GetUserByEmail(email)
+		if err != nil {
+			return nil, fmt.Errorf("could not get slack user for %s: %w", member, err)
+		}
+		users[member] = user{
+			KerberosId: member,
+			TeamName:   config.TeamName,
+			SlackId:    slackUser.ID,
+		}
+	}
+
+	for githubId, kerberosId := range gtk {
+		userInfo, exists := users[kerberosId]
+		if exists {
+			userInfo.GithubId = githubId
+			users[kerberosId] = userInfo
+		}
+	}
+
+	var usersMissingGithubId []string
+	for _, userInfo := range users {
+		if userInfo.GithubId == "" {
+			usersMissingGithubId = append(usersMissingGithubId, userInfo.KerberosId)
+		}
+	}
+	if len(usersMissingGithubId) > 0 {
+		return nil, fmt.Errorf("no githubId found for user(s): %v", usersMissingGithubId)
+	}
+
+	return users, nil
+}
+
+func messageUser(user user, slackClient slackClient) error {
```

Comment:
This is just a suggestion: I would split this function into smaller chunks. It seems to be quite long.

Reply:
It is quite long, but its 90% simple boiler plate that the slack client requires to send a message. I think breaking it up would just make it even more time consuming to read.